src/uwtools/config/validator.py

```diff
@@ -47,37 +47,37 @@ def validate(schema: dict, config: dict) -> bool:
 
 
 def validate_internal(
-    schema_name: str, config: Union[dict, YAMLConfig, Optional[Path]] = None
+    schema_name: str, config: Optional[Union[dict, YAMLConfig, Path]] = None
 ) -> None:
     """
     Validate a config against a uwtools-internal schema.
 
-    :param config: The config to validate.
     :param schema_name: Name of uwtools schema to validate the config against.
+    :param config: The config to validate.
     :raises: UWConfigError if config fails validation.
     """
 
     log.info("Validating config against internal schema %s", schema_name)
     schema_file = get_schema_file(schema_name)
     log.debug("Using schema file: %s", schema_file)
-    if not validate_external(config=config, schema_file=schema_file):
-        raise UWConfigError("YAML validation errors")
+    validate_external(config=config, schema_file=schema_file)
 
 
 def validate_external(
     schema_file: Path, config: Union[dict, YAMLConfig, Optional[Path]] = None
```

Comment:
Any reason not to have this function signature match `validate_internal`?

Reply:
Yes: `validate_internal` only knows the schema _name_; the path to the file is determined programmatically, based on `uwtools` internal conventions, via `importlib`, because the built-in `.jsonschema` files are package resources. So, `validate_internal` operates on a name, and `validate_external` operates on a path (file).

Reply:
I apparently wrote a comment here and lost it another of the many tabs. :(

I was too general in my original comment. I was really just suggesting that we might follow the same pattern with the type hint for `config`:

```
config: Optional[Union[dict, YAMLConfig, Path]]
```

Reply:
Oh, I see now: Yes, the `Optional` could move outside for consistency. Doing that now. Thanks.